phpunit/html/WP_HTML_Tag_Processor_Test.php

```diff
@@ -217,6 +217,22 @@ public function test_next_tag_should_return_false_for_a_non_existing_tag() {
 		$this->assertFalse( $p->next_tag( 'p' ), 'Querying a non-existing tag did not return false' );
 	}
 
+	/**
+	 * @covers next_tag
+	 * @covers is_tag_closer
+	*/
+	public function test_next_tag_should_stop_on_closers_only_when_requested() {
+		$p = new WP_HTML_Tag_Processor( '<div><img /></div>' );
+		$this->assertTrue( $p->next_tag( [ 'tag_name' => 'div' ] ), 'Did not find desired tag opener' );
+		$this->assertFalse( $p->next_tag( [ 'tag_name' => 'div' ] ), 'Visited an unwanted tag, a tag closer' );
+
+		$p = new WP_HTML_Tag_Processor( '<div><img /></div>' );
+		$p->next_tag( [ 'tag_name' => 'div', 'tag_closers' => 'visit' ] );
+		$this->assertFalse( $p->is_tag_closer(), 'Indicated a tag opener is a tag closer' );
+		$this->assertTrue( $p->next_tag( [ 'tag_name' => 'div', 'tag_closers' => 'visit' ] ), 'Did not stop at desired tag closer' );
+		$this->assertTrue( $p->is_tag_closer(), 'Indicated a tag closer is a tag opener' );
```

Comment:
I'd like to see some tests to confirm the behavior of the other methods in the tag closer state, e.g. `update_attribute()`, `__toString()`, etc

Reply:
added a test to confirm that ops are inert inside closed tags